### PR TITLE
WIP: Add handler and operator Dockerfiles for openshift

### PR DESCRIPTION
**Pull request by yboaron**
**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: build/Dockerfile.openshift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
+WORKDIR /go/src/github.com/nmstate/kubernetes-nmstate
+COPY . .
+RUN GO111MODULE=on go build --mod=vendor -o build/_output/bin/kubernetes-nmstate cmd/manager/main.go
+
+FROM centos:8
```
Comment (bcrochet) [inline build/Dockerfile.openshift:6]: This should probably be the ubi8-minimal like the operator if it's for OpenShift.

File: build/Dockerfile.openshift [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
```
[the remark this answers is not in the document]
Comment (phoracek) [reply build/Dockerfile.openshift:1]: We have to use 1.13 for our controller-runtime version

File: build/Dockerfile.openshift [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR /go/src/github.com/nmstate/kubernetes-nmstate
+COPY . .
+RUN GO111MODULE=on go build --mod=vendor -o build/_output/bin/kubernetes-nmstate cmd/manager/main.go
+
+FROM centos:8
+
+RUN yum install -b -y dnf-plugins-core && \
+#    yum copr enable -y networkmanager/NetworkManager-1.22 && \
```
[the remark this answers is not in the document]
Comment (phoracek) [reply build/Dockerfile.openshift:9]: We can install all dependencies with:

```
RUN \
    microdnf -y update && \
    microdnf -y install \
        nmstate \
        iputils \
        iproute && \
    microdnf clean all
```

File: build/Dockerfile.openshift [hunk 4]
```diff
@@ -0,0 +1,18 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
+WORKDIR /go/src/github.com/nmstate/kubernetes-nmstate
+COPY . .
+RUN GO111MODULE=on go build --mod=vendor -o build/_output/bin/kubernetes-nmstate cmd/manager/main.go
```
[the remark this answers is not in the document]
Comment (qinqon) [reply build/Dockerfile.openshift:4]: That's complicated our makefile do the operator sdk build that's a monolithic thing

File: build/Dockerfile.openshift [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#    yum copr enable -y networkmanager/NetworkManager-1.22 && \
+#    yum copr enable -y nmstate/nmstate-0.2 && \
+    yum install -b -y nmstate iproute iputils && \
+    yum remove -y dnf-plugins-core && \
+    yum clean all
+
+COPY --from=builder /go/src/github.com/nmstate/kubernetes-nmstate/build/_output/bin/kubernetes-nmstate  /usr/bin/
+COPY --from=builder /go/src/github.com/nmstate/kubernetes-nmstate/build/bin   /usr/bin/
+
+ENTRYPOINT ["/usr/bin/kubernetes-nmstate"]
```
Comment (phoracek) [inline build/Dockerfile.openshift:18]: Don't we need to `LABEL` the dockerfile? https://github.com/openshift/cluster-network-operator/blob/master/Dockerfile.rhel7#L16

Comment (bcrochet) [reply build/Dockerfile.openshift:18]: I will need to find it, but I think I saw something about needing to wait to do that until something happens on the CVO side maybe? Let me see if I can dig up that reference.

File: build/Dockerfile.openshift (hunk 1 above)
Comment (yboaron) [reply build/Dockerfile.openshift]: Done

File: build/Dockerfile.openshift (hunk 4 above)
[the remark this answers is not in the document]
Comment (yboaron) [reply build/Dockerfile.openshift:4]: I also think, we should keep it simple (go build --mod-vendor ... )

File: build/Dockerfile.openshift (hunk 2 above)
[the remark this answers is not in the document]
Comment (yboaron) [reply build/Dockerfile.openshift]: Copy-paste error. Done

File: build/Dockerfile.openshift (hunk 5 above)
Comment (bcrochet) [reply build/Dockerfile.openshift:20]: I can't find it. Ignore me. :)

**Comment by bcrochet** [comment]
> /lgtm
> /approve
> /hold
> 
> Have you verified that we are able to build these images?

Locally, I got 'No package matches nmstate'

**Comment by phoracek** [comment]
@bcrochet on D/S, we included these repos while building the image:

```
- rhel-8-for-x86_64-baseos-rpms
- rhel-8-for-x86_64-appstream-rpms
```

Maybe we need to get our hands on base-os RPMs here as well? Or use full-blown RHEL as the base-image?

**Comment by yboaron** [comment]
Openshift Docker files pushed to U/S repo, so we can close this PR.

**Pull request closed without merging**